app/src/main/java/com/github/campus_capture/bootcamp/map/MapScheduler.java

```diff
@@ -129,16 +131,25 @@ public void run() {
      * @param backend the back-end to be used
      * @param upper the fragment above
      */
-    public MapScheduler(View view, FirebaseInterface backend, MapsFragment upper)
+    public MapScheduler(View view, BackendInterface backend, MapsFragment upper)
     {
-        firebaseInterface = backend;
+        backendInterface = backend;
         this.upper = upper;
         scheduledTaskHandler = new Handler();
         zoneText = view.findViewById(R.id.currentZoneText);
         attackButton = view.findViewById(R.id.attackButton);
         defendButton = view.findViewById(R.id.defendButton);
         timerButton = view.findViewById(R.id.timerButton);
-        zoneState = firebaseInterface.getCurrentZoneOwners();
+
+        // TODO is not setting zoneState immediately a problem ?
```

Comment:
Nope! This was intended this way, to prevent many concurrent calls to the back-end.